File: Week 1/mario-less.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // Запрашиваем высоту пирамиды, при условии [1;8]
    int n;
    while (true)
    {
        n = get_int("Height: ");
        if (n > 0 && n < 9)
        {
            break;
        }
    }
    // Цикл количества строк и столбцов
    for (int i = 0; i < n; i++)
    {
        // Цикл точки в строке
        for (int j = 0; j < n - i - 1; j++)
        {
            printf(" ");
        }
        // Цикл #
        for (int z = 0; z < i + 1; z++)
        {
            printf("#");
        }
        printf("\n");
    }
}

File: Week 6/mario less.py
from cs50 import get_int


def main():
    height = get_height()
    for col in range(1, height + 1):
        print_symb(height, col)
# Print symbols


def print_symb(n, col):
    print(" " * (n - col) + "#" * col, end="\n")
# Input validation


def get_height():
    while True:
        h = get_int("Height: ")
        if 1 <= h <= 8:
            break
    return h


main()